tools/gcp_init/installer.py

```diff
@@ -0,0 +1,680 @@
+    """Gets list of enabled gcloud services.
+
+    Returns:
+      string: JSON encoded list of enabled gcloud services
+    """
+    log.info(('getting enabled services '
+              'for project: \'{0:s}\'').format(self.project))
+    self.svcs_list = None
+    self.svcs_list_full = None
+    with fab.hide('running'):
+      status = fab.local(self.__BuildCmd(verb='services list'), capture=True)
+
+      try:
+        self.svcs_list_full = json.loads(status.stdout)
+      except ValueError as e:
+        print('Error parsing json returned by gcloud cli tool: {0:s}'.format(e))
+        traceback.print_tb(sys.exc_info()[2])
+
+      self.svcs_list = list()
+
+      for entry in self.svcs_list_full:
+        self.svcs_list.append(entry['serviceName'])
+
+  def GetComputeInstances(self):
+    """Gets list of gcloud compute instances.
+
+    Returns:
+      string: JSON encoded data structure of gcloud compute instances
+    """
+    log.info(('getting compute instances '
+              'for project: \'{0:s}\'').format(self.project))
+    self.compute_instances = None
+    self.compute_instances_full = None
+
+    with fab.hide('running'):
+      status = fab.local(self.__BuildCmd(verb='compute instances list'),
+                         capture=True)
+
+      try:
+        self.compute_instances_full = json.loads(status.stdout)
+      except ValueError as e:
+        print('Error parsing json returned by gcloud cli tool: {0:s}'.format(e))
+        traceback.print_tb(sys.exc_info()[2])
+
+      self.compute_instances = list()
+
+      for entry in self.compute_instances_full:
+        self.compute_instances.append(entry['name'])
+
+  def GetIamPolicy(self):
+    """Gets IAM policy for gcloud project.
+
+    Returns:
+      string: JSON encoded iam policy
+    """
+    log.info(('getting iam policy for project: '
+              '\'{0:s}\'').format(self.project))
+    with fab.hide('running'):
+      status = fab.local(self.__BuildCmd(verb=('projects get-iam-policy {0:s}'
+                                              ).format(self.project)),
+                         capture=True)
+
+      try:
+        self.iam_policy = json.loads(status.stdout)
+      except ValueError as e:
+        print('Error parsing json returned by gcloud cli tool: {0:s}'.format(e))
+        traceback.print_tb(sys.exc_info()[2])
+
+  def SetProject(self):
+    """Sets project name for gcloud cmd object."""
+    log.info('setting project to: \'{0:s}\''.format(self.project))
+    with fab.hide('running'):
+      status = fab.local(('gcloud config set '
+                          'project {0:s}').format(self.project),
+                         capture=True)
+      if not status.stderr.startswith('Updated property [core/project].'):
+        log.error('error setting project. got: \'{0:s}\''.format(status.stderr))
+
+  def CreateServiceAccount(self):
+    """Gets IAM policy for gcloud project."""
+    self.svc_account_keys_file = ('../../../../../etc/'
```

Comment:
This seems super brittle. What is this relative to?

Reply:
Agreed, I need to make this a cli option to specify the path to the service account keys file.